restapi/src/main/java/io/stargate/web/resources/v2/schemas/IndexesResource.java

```diff
@@ -0,0 +1,280 @@
+      })
+  public Response addIndex(
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam("X-Cassandra-Token")
+          String token,
+      @ApiParam(value = "Name of the keyspace to use for the request.", required = true)
+          @PathParam("keyspaceName")
+          final String keyspaceName,
+      @ApiParam(required = true) @NotNull final IndexAdd indexAdd,
+      @Context HttpServletRequest request) {
+    return RequestHandler.handle(
+        () -> {
+          Map<String, String> allHeaders = getAllHeaders(request);
+          AuthenticatedDB authenticatedDB = db.getDataStoreForToken(token, allHeaders);
+
+          String tableName = indexAdd.getTable();
+          String columnName = indexAdd.getColumn();
+
+          Keyspace keyspace = authenticatedDB.getDataStore().schema().keyspace(keyspaceName);
+          if (keyspace == null) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                .entity(
+                    new Error(
+                        String.format("Keyspace '%s' not found.", keyspaceName),
+                        Response.Status.BAD_REQUEST.getStatusCode()))
+                .build();
+          }
+
+          try {
+            final Table tableMetadata = authenticatedDB.getTable(keyspaceName, tableName);
+            final Column col = tableMetadata.column(columnName);
+            if (col == null) {
+              return Response.status(Response.Status.NOT_FOUND)
+                  .entity(
+                      new Error(
+                          String.format("Column '%s' not found in table.", columnName),
+                          Response.Status.NOT_FOUND.getStatusCode()))
+                  .build();
+            }
+          } catch (NotFoundException e) {
+            return Response.status(Response.Status.NOT_FOUND)
+                .entity(
+                    new Error(
+                        String.format("Table '%s' not found in keyspace.", tableName),
+                        Response.Status.NOT_FOUND.getStatusCode()))
+                .build();
+          }
+
+          createIndex(keyspaceName, indexAdd, authenticatedDB, tableName, columnName);
+
+          return Response.status(Response.Status.CREATED).entity(new SuccessResponse()).build();
+        });
+  }
+
+  private void createIndex(
+      String keyspaceName,
+      IndexAdd indexAdd,
+      AuthenticatedDB authenticatedDB,
+      String tableName,
+      String columnName)
+      throws UnauthorizedException, InterruptedException, java.util.concurrent.ExecutionException {
+    db.getAuthorizationService()
+        .authorizeSchemaWrite(
+            authenticatedDB.getAuthenticationSubject(),
+            keyspaceName,
+            tableName,
+            Scope.CREATE,
+            SourceAPI.REST);
+
+    boolean indexKeys = false;
```

Comment:
Um... I could simplify the lines 147 to 169 as:
```
boolean indexKeys = indexAdd.getKind() == KEYS;
boolean indexEntries = indexAdd.getKind() == ENTRIES;
boolean indexValues = indexAdd.getKind() == VALUES;
boolean indexFull = indexAdd.getKind() == FULL;
```

Also, I don't think there would be situation where the default clause would be reached...